File: .sdk-repos/orchestration-cluster-api-python/examples/group.py
# Compilable usage examples for group management operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    ClientId,
    GroupCreateRequest,
    GroupId,
    GroupSearchQueryRequest,
    GroupUpdateRequest,
    MappingRuleId,
    MappingRuleSearchQueryRequest,
    RoleSearchQueryRequest,
    Unset,
    Username,
)


# region CreateGroup
def create_group_example(group_id: GroupId) -> None:
    client = CamundaClient()

    result = client.create_group(
        data=GroupCreateRequest(group_id=group_id, name="Engineering"),
    )

    print(f"Group: {result.group_id}")
# endregion CreateGroup


# region GetGroup
def get_group_example(group_id: GroupId) -> None:
    client = CamundaClient()

    result = client.get_group(group_id=group_id)

    print(f"Group: {result.name}")
# endregion GetGroup


# region SearchGroups
def search_groups_example() -> None:
    client = CamundaClient()

    result = client.search_groups(
        data=GroupSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for group in result.items:
            print(f"Group: {group.name}")
# endregion SearchGroups


# region UpdateGroup
def update_group_example(group_id: GroupId) -> None:
    client = CamundaClient()

    client.update_group(
        group_id=group_id,
        data=GroupUpdateRequest(name="engineering-team"),
    )
# endregion UpdateGroup


# region DeleteGroup
def delete_group_example(group_id: GroupId) -> None:
    client = CamundaClient()

    client.delete_group(group_id=group_id)
# endregion DeleteGroup


# region AssignUserToGroup
def assign_user_to_group_example(group_id: GroupId, username: Username) -> None:
    client = CamundaClient()

    client.assign_user_to_group(
        group_id=group_id,
        username=username,
    )
# endregion AssignUserToGroup


# region UnassignUserFromGroup
def unassign_user_from_group_example(group_id: GroupId, username: Username) -> None:
    client = CamundaClient()

    client.unassign_user_from_group(
        group_id=group_id,
        username=username,
    )
# endregion UnassignUserFromGroup


# region AssignClientToGroup
def assign_client_to_group_example(group_id: GroupId, client_id: ClientId) -> None:
    client = CamundaClient()

    client.assign_client_to_group(
        group_id=group_id,
        client_id=client_id,
    )
# endregion AssignClientToGroup


# region UnassignClientFromGroup
def unassign_client_from_group_example(group_id: GroupId, client_id: ClientId) -> None:
    client = CamundaClient()

    client.unassign_client_from_group(
        group_id=group_id,
        client_id=client_id,
    )
# endregion UnassignClientFromGroup


# region AssignMappingRuleToGroup
def assign_mapping_rule_to_group_example(group_id: GroupId, mapping_rule_id: MappingRuleId) -> None:
    client = CamundaClient()

    client.assign_mapping_rule_to_group(
        group_id=group_id,
        mapping_rule_id=mapping_rule_id,
    )
# endregion AssignMappingRuleToGroup


# region UnassignMappingRuleFromGroup
def unassign_mapping_rule_from_group_example(group_id: GroupId, mapping_rule_id: MappingRuleId) -> None:
    client = CamundaClient()

    client.unassign_mapping_rule_from_group(
        group_id=group_id,
        mapping_rule_id=mapping_rule_id,
    )
# endregion UnassignMappingRuleFromGroup


# region SearchUsersForGroup
def search_users_for_group_example(group_id: GroupId) -> None:
    client = CamundaClient()

    result = client.search_users_for_group(
        group_id=group_id,
    )

    if not isinstance(result.items, Unset):
        for user in result.items:
            print(f"User: {user.username}")
# endregion SearchUsersForGroup


# region SearchClientsForGroup
def search_clients_for_group_example(group_id: GroupId) -> None:
    client = CamundaClient()

    result = client.search_clients_for_group(
        group_id=group_id,
    )

    if not isinstance(result.items, Unset):
        for c in result.items:
            print(f"Client: {c.client_id}")
# endregion SearchClientsForGroup


# region SearchRolesForGroup
def search_roles_for_group_example(group_id: GroupId) -> None:
    client = CamundaClient()

    result = client.search_roles_for_group(
        group_id=group_id,
        data=RoleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for role in result.items:
            print(f"Role: {role.name}")
# endregion SearchRolesForGroup


# region SearchMappingRulesForGroup
def search_mapping_rules_for_group_example(group_id: GroupId) -> None:
    client = CamundaClient()

    result = client.search_mapping_rules_for_group(
        group_id=group_id,
        data=MappingRuleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for rule in result.items:
            print(f"Mapping rule: {rule.mapping_rule_id}")
# endregion SearchMappingRulesForGroup

File: .sdk-repos/orchestration-cluster-api-csharp/examples/Group.cs
// Compilable usage examples for group management operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class GroupExamples
{
    #region CreateGroup
    // <CreateGroup>
    public static async Task CreateGroupExample(GroupId groupId)
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateGroupAsync(new GroupCreateRequest
        {
            GroupId = groupId,
            Name = "Engineering",
        });

        Console.WriteLine($"Group key: {result.GroupId}");
    }
    // </CreateGroup>
    #endregion CreateGroup

    #region GetGroup

    // <GetGroup>
    public static async Task GetGroupExample(GroupId groupId)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetGroupAsync(groupId);
        Console.WriteLine($"Group: {result.Name}");
    }
    // </GetGroup>
    #endregion GetGroup

    #region SearchGroups

    // <SearchGroups>
    public static async Task SearchGroupsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchGroupsAsync(new GroupSearchQueryRequest());

        foreach (var group in result.Items)
        {
            Console.WriteLine($"Group: {group.Name}");
        }
    }
    // </SearchGroups>
    #endregion SearchGroups

    #region UpdateGroup

    // <UpdateGroup>
    public static async Task UpdateGroupExample(GroupId groupId)
    {
        using var client = CamundaClient.Create();

        await client.UpdateGroupAsync(groupId, new GroupUpdateRequest
        {
            Name = "engineering-team",
        });
    }
    // </UpdateGroup>
    #endregion UpdateGroup

    #region DeleteGroup

    // <DeleteGroup>
    public static async Task DeleteGroupExample(GroupId groupId)
    {
        using var client = CamundaClient.Create();

        await client.DeleteGroupAsync(groupId);
    }
    // </DeleteGroup>
    #endregion DeleteGroup

    #region AssignUserToGroup

    // <AssignUserToGroup>
    public static async Task AssignUserToGroupExample(GroupId groupId, Username username)
    {
        using var client = CamundaClient.Create();

        await client.AssignUserToGroupAsync(groupId, username);
    }
    // </AssignUserToGroup>
    #endregion AssignUserToGroup

    #region UnassignUserFromGroup

    // <UnassignUserFromGroup>
    public static async Task UnassignUserFromGroupExample(GroupId groupId, Username username)
    {
        using var client = CamundaClient.Create();

        await client.UnassignUserFromGroupAsync(groupId, username);
    }
    // </UnassignUserFromGroup>
    #endregion UnassignUserFromGroup

    #region AssignClientToGroup

    // <AssignClientToGroup>
    public static async Task AssignClientToGroupExample(GroupId groupId, ClientId clientId)
    {
        using var client = CamundaClient.Create();

        await client.AssignClientToGroupAsync(groupId, clientId);
    }
    // </AssignClientToGroup>
    #endregion AssignClientToGroup

    #region UnassignClientFromGroup

    // <UnassignClientFromGroup>
    public static async Task UnassignClientFromGroupExample(GroupId groupId, ClientId clientId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignClientFromGroupAsync(groupId, clientId);
    }
    // </UnassignClientFromGroup>
    #endregion UnassignClientFromGroup

    #region AssignMappingRuleToGroup

    // <AssignMappingRuleToGroup>
    public static async Task AssignMappingRuleToGroupExample(GroupId groupId, MappingRuleId mappingRuleId)
    {
        using var client = CamundaClient.Create();

        await client.AssignMappingRuleToGroupAsync(groupId, mappingRuleId);
    }
    // </AssignMappingRuleToGroup>
    #endregion AssignMappingRuleToGroup

    #region UnassignMappingRuleFromGroup

    // <UnassignMappingRuleFromGroup>
    public static async Task UnassignMappingRuleFromGroupExample(GroupId groupId, MappingRuleId mappingRuleId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignMappingRuleFromGroupAsync(groupId, mappingRuleId);
    }
    // </UnassignMappingRuleFromGroup>
    #endregion UnassignMappingRuleFromGroup

    #region SearchUsersForGroup

    // <SearchUsersForGroup>
    public static async Task SearchUsersForGroupExample(GroupId groupId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchUsersForGroupAsync(
            groupId,
            new GroupUserSearchQueryRequest());

        foreach (var user in result.Items)
        {
            Console.WriteLine($"User: {user.Username}");
        }
    }
    // </SearchUsersForGroup>
    #endregion SearchUsersForGroup

    #region SearchClientsForGroup

    // <SearchClientsForGroup>
    public static async Task SearchClientsForGroupExample(GroupId groupId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchClientsForGroupAsync(
            groupId,
            new GroupClientSearchQueryRequest());

        foreach (var c in result.Items)
        {
            Console.WriteLine($"Client: {c.ClientId}");
        }
    }
    // </SearchClientsForGroup>
    #endregion SearchClientsForGroup

    #region SearchRolesForGroup

    // <SearchRolesForGroup>
    public static async Task SearchRolesForGroupExample(GroupId groupId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchRolesForGroupAsync(
            groupId,
            new RoleSearchQueryRequest());

        foreach (var role in result.Items)
        {
            Console.WriteLine($"Role: {role.Name}");
        }
    }
    // </SearchRolesForGroup>
    #endregion SearchRolesForGroup

    #region SearchMappingRulesForGroup

    // <SearchMappingRulesForGroup>
    public static async Task SearchMappingRulesForGroupExample(GroupId groupId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchMappingRulesForGroupAsync(
            groupId,
            new MappingRuleSearchQueryRequest());

        foreach (var rule in result.Items)
        {
            Console.WriteLine($"Mapping rule: {rule.MappingRuleId}");
        }
    }
    // </SearchMappingRulesForGroup>
    #endregion SearchMappingRulesForGroup
}
